WordPressVIPMinimum/Sniffs/Security/ProperEscapingFunctionSniff.php

```diff
@@ -47,47 +69,40 @@ public function register() {
 	 */
 	public function process_token( $stackPtr ) {
 
-		if ( in_array( $this->tokens[ $stackPtr ]['content'], $this->escaping_functions, true ) === false ) {
+		if ( isset( $this->escaping_functions[ $this->tokens[ $stackPtr ]['content'] ] ) === false ) {
 			return;
 		}
 
-		$function_name = $this->tokens[ $stackPtr ]['content'];
+		$html = $this->phpcsFile->findPrevious( $this->echo_or_concat_tokens, $stackPtr - 1, null, true );
 
-		$echo_or_string_concat = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $stackPtr - 1, null, true );
+		// Use $textStringTokens b/c heredoc and nowdoc tokens shouldn't be matched anyways.
+		if ( $html === false || isset( Tokens::$textStringTokens[ $this->tokens[ $html ]['code'] ] ) === false ) {
+			return;
+		}
 
-		if ( $this->tokens[ $echo_or_string_concat ]['code'] === T_ECHO ) {
-			// Very likely inline HTML with <?php tag.
-			$php_open = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $echo_or_string_concat - 1, null, true );
+		$function_name = $this->tokens[ $stackPtr ]['content'];
 
-			if ( $this->tokens[ $php_open ]['code'] !== T_OPEN_TAG ) {
-				return;
-			}
+		$data = [ $function_name ];
 
-			$html = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $php_open - 1, null, true );
+		$content = $this->tokens[ $html ]['content'];
 
-			if ( $this->tokens[ $html ]['code'] !== T_INLINE_HTML ) {
-				return;
-			}
-		} elseif ( $this->tokens[ $echo_or_string_concat ]['code'] === T_STRING_CONCAT ) {
-			// Very likely string concatenation mixing strings and functions/variables.
-			$html = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $echo_or_string_concat - 1, null, true );
+		if ( isset( Tokens::$stringTokens[ $this->tokens[ $html ]['code'] ] ) === true ) {
+			$content = Sniff::strip_quotes( $content );
```

Comment:
👍 

The only thing is, now this has been changed, the code in the `attr_expects_url()` and the `is_html_attr()` functions needs to be updated too.

The fact that the tests aren't failing on it, only highlights the fact that tests for some parts of those functions are missing (or more likely even: that the author of the sniff couldn't get the tests working as some of the code in those functions is actually faulty).

Suggested unit tests:
```php
// These should throw the `htmlAttrNotByEscHTML` error.
echo '<div class=\'' . esc_html($class) . '\'>'; // Error.
echo "<div class=\"" . esc_html($class) . '">'; // Error.
echo "<div $someAttribute class=\"" . esc_html($class) . '">'; // Error.

// These should throw the `hrefSrcEscUrl` error.
echo '<a href=\'' . esc_html($url) . '\'>'; // Error.
echo "<img src=\"" . esc_html($src) . '"/>'; // Error.
echo "<div $someAttributeName-url=\"" . esc_html($url) . '">'; // Error.
```

Reply:
Also:

```php
echo '<a href=', esc_html($url), '>';
```